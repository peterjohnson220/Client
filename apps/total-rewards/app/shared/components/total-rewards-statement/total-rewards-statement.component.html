<div *ngIf="statement" class="container statement mb-4 rounded" [ngClass]="fontSizeCssClass + ' ' + fontFamilyCssClass">
  <div *ngFor="let employee of employees; trackBy: trackByFn">
    <div class="d-flex justify-content-end">
      <pf-effective-date 
        [mode]="mode"
        [effectiveDate]="statement.EffectiveDate"
        (onDateChange)="handleEffectiveDateChange($event)">
      </pf-effective-date>
    </div>
    <div *ngFor="let page of statement.Pages; trackBy: trackByFn" class="pb-4 trs-page mt-n4">
      <section *ngFor="let section of page.Sections; trackBy: trackByFn" class="row vertical-divider trs-section">
        <div *ngFor="let column of section.Columns; trackBy: trackByFn" [ngClass]="'col-' + column?.Layout?.Width" class="trs-column">
          <div class="row columnRow h-100">
            <div *ngFor="let control of column.Controls; trackBy: trackByFn" [ngClass]="'col-' + control?.Layout?.Width + ' control-type-' + control.ControlType">
              <div *ngIf="control.ControlType === controlType.RichTextEditor">
                <pf-trs-rich-text-control
                  [controlData]="control"
                  [mode]="mode"
                  [employeeRewardsData]="employeeRewardsData"
                  (onTitleChange)="handleOnControlTitleChange($event)"
                  (onContentChange)="handleOnRichTextControlContentChange($event)"
                  class="p-1"></pf-trs-rich-text-control>
              </div>

              <div *ngIf="control.ControlType === controlType.Image">
                <pf-trs-image-control
                  [controlData]="control"
                  [statementId]="statement.StatementId"
                  [mode]="mode"
                  (saveImage)="handleSaveImage($event)"
                  (removeImage)="handleRemoveImage($event)"
                  class="p-1"></pf-trs-image-control>
              </div>

              <div *ngIf="control.ControlType === controlType.Calculation">
                <pf-trs-calculation-control
                  [controlData]="control"
                  [mode]="mode"
                  [employeeRewardsData]="employeeRewardsData"
                  (onTitleChange)="handleOnControlTitleChange($event)"
                  (onCompFieldTitleChange)="handleOnCalculationControlCompFieldTitleChange($event)"
                  (onUpdateSummaryTitleChange)="handleOnCalculationControlSummaryTitleChange($event)"
                  (onCompFieldRemoved)="handleOnCalculationControlCompFieldRemoved($event)"
                  (onCompFieldAdded)="handleOnCalculationControlCompFieldAdded($event)"></pf-trs-calculation-control>
              </div>

              <div *ngIf="control.ControlType === controlType.Chart">
                <pf-trs-chart-control
                  [controlData]="control"
                  [mode]="mode"
                  [employeeRewardsData]="employeeRewardsData"
                  [calculationControls]="calculationControls"
                  [chartColors]="statement.Settings.ChartColors"
                  (onTitleChange)="handleOnControlTitleChange($event)"
                  (settingsClick)="handleChartControlSettingsClick()"
                  class="p-1"></pf-trs-chart-control>
              </div>

              <div *ngIf="control.ControlType === controlType.Title">
                <pf-trs-title-control
                  [controlData]="control"
                  [mode]="mode"
                  [employeeRewardsData]="employeeRewardsData"
                  (titleChange)="handleOnControlTitleChange($event)"
                  class="p-1"></pf-trs-title-control>
              </div>

              <div *ngIf="control.ControlType === controlType.CalculationSummary">
                <pf-trs-summary-control
                  [mode]="mode"
                  [controlData]="control"
                  [employeeRewardsData]="employeeRewardsData"
                  [calculationControls]="calculationControls"
                  (onTitleChange)="handleOnControlTitleChange($event)"
                  class="p-1"></pf-trs-summary-control>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div class="page-break"></div>
  </div>
</div>
