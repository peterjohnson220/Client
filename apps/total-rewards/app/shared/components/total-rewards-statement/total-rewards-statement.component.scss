@import "app-colors";

@media screen {
  .statement {
    background: white;
    width: 8.5in;
    max-height: 11in;
    margin-top: 50px;
    overflow-x: hidden; // keep effective date picker from leaking in preview
  }
}

.page-break {
  page-break-after: always;
}

.row.vertical-divider > div[class*="col-"] {
  border-left: 1px solid $trs-canvas-background-color;
  border-right: 1px solid $trs-canvas-background-color;
}

// hide the middle divider for the top row
.row.vertical-divider:first-child .trs-column {
  border-width: 0;
}

.control-type-5 {
  align-self: center;
}

@media print {
  .page-break	{
    display: block;
    page-break-before: always;
  }

  .column {
    border: none;
  }
}

@media screen {
  .page-break	{
    display: block;
    page-break-before: always;
  }
}
