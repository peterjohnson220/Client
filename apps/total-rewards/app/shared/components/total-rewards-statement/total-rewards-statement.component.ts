import {Component, Input, OnInit, OnDestroy, Output, EventEmitter, ChangeDetectionStrategy} from '@angular/core';
import { ActivatedRoute } from '@angular/router';

import { Subscription } from 'rxjs';

import {
  Statement,
  TotalRewardsControlEnum,
  UpdateFieldOverrideNameRequest,
  UpdateFieldVisibilityRequest,
  UpdateStringPropertyRequest,
  UpdateTitleRequest,
  EmployeeRewardsData,
  StatementModeEnum,
  CalculationControl,
  SaveImageRequest,
  DeleteImageRequest
} from '../../models';

@Component({
  selector: 'pf-total-rewards-statement',
  templateUrl: './total-rewards-statement.component.html',
  styleUrls: ['./total-rewards-statement.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TotalRewardsStatementComponent implements OnInit, OnDestroy {

  @Input() statement: Statement;
  @Input() mode: StatementModeEnum;
  @Input() employeeRewardsData: EmployeeRewardsData;

  // Common Outputs
  @Output() onControlTitleChange: EventEmitter<UpdateTitleRequest> = new EventEmitter();

  // Calculation Outputs
  @Output() onCalculationControlCompFieldTitleChange: EventEmitter<UpdateFieldOverrideNameRequest> = new EventEmitter();
  @Output() onCalculationControlSummaryTitleChange: EventEmitter<UpdateTitleRequest> = new EventEmitter();
  @Output() onCalculationControlCompFieldRemoved: EventEmitter<UpdateFieldVisibilityRequest> = new EventEmitter();
  @Output() onCalculationControlCompFieldAdded: EventEmitter<UpdateFieldVisibilityRequest> = new EventEmitter();

  // Rich Text Outputs
  @Output() onRichTextControlContentChange: EventEmitter<UpdateStringPropertyRequest> = new EventEmitter<UpdateStringPropertyRequest>();

  // Chart Control Outputs
  @Output() onChartControlToggleSettingsPanelClick = new EventEmitter();

  // Image Control Outputs
  @Output() onSaveImage: EventEmitter<SaveImageRequest> = new EventEmitter();
  @Output() onRemoveImage: EventEmitter<DeleteImageRequest> = new EventEmitter();

  controlType = TotalRewardsControlEnum;
  statementModeEnum = StatementModeEnum;

  // check statement.Settings.FontSize and return small-font-size | medium-font-size | large-font-size | ''
  get fontSizeCssClass(): string {
    if (this.statement && this.statement.Settings && this.statement.Settings.FontSize) {
      return this.statement.Settings.FontSize.toLowerCase() + '-font-size';
    }
    return '';
  }

  // check statement.Settings.FontFamily and return 'arial-font-family' | 'times-new-roman-font-family', etc
  get fontFamilyCssClass(): string {
    if (this.statement && this.statement.Settings && this.statement.Settings.FontFamily) {
      return this.statement.Settings.FontFamily.toLowerCase().replace(/ /g, '-') + '-font-family';
    }
    return '';
  }

  get calculationControls(): CalculationControl[] {
    if (!this.statement) {
      return [];
    }

    const calcControls = [];
    this.statement.Pages.forEach(p => p.Sections.forEach(s => s.Columns.forEach(c => c.Controls.forEach(control => {
      if (control.ControlType === TotalRewardsControlEnum.Calculation) {
        calcControls.push(control);
      }
    }))));

    return calcControls;
  }

  employeeData = [
    {
      name: 'Employee 1',
      compensationData: [
        { value: 55, category: 'Base' },
        { value: 10, category: 'Bonus' },
        { value: 20, category: 'Healthcare' },
        { value: 5, category: '401K Match' }
        ],
      logoPath: 'https://images-na.ssl-images-amazon.com/images/I/41yf7iS-BML._SX355_.jpg'
    },
    {
      name: 'Employee 2',
      compensationData: [
        { value: 40000, category: 'Base' },
        { value: 20000, category: 'Bonus' },
        { value: 5000, category: 'STI' },
        { value: 10000, category: 'LTI' },
        { value: 20000, category: 'Medical_Insurance' },
        { value: 20000, category: '401K_Savings_Match' }
      ],
      logoPath: 'https://vignette.wikia.nocookie.net/theoffice/images/0/02/Michael_Scott.jpg/revision/latest?cb=20170701090332'
    }
  ];

  private pageCountSubscription: Subscription;
  pageCount = 1;
  employeeArray = [];

  constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    // This is for POC employees/pages
    this.pageCountSubscription = this.route.queryParams.subscribe(params => {
      if (params['pages']) {
        this.pageCount = params['pages'];
      }
      for (let i = 0; i < this.pageCount; i++) {
        this.employeeArray.push(this.employeeData[i % 2]);
      }
    });
  }

  ngOnDestroy() {
    this.pageCountSubscription.unsubscribe();
  }

  // track which item each ngFor is on, which no longer necessitates destroying/creating all components in state changes and improves perf significantly
  trackByFn(index: number, item: any) {
    return index;
  }

  // Common pass through methods
  handleOnControlTitleChange(event) {
    this.onControlTitleChange.emit(event);
  }

  // Calculation pass through methods
  handleOnCalculationControlCompFieldTitleChange(event) {
    this.onCalculationControlCompFieldTitleChange.emit(event);
  }

  handleOnCalculationControlSummaryTitleChange(event) {
    this.onCalculationControlSummaryTitleChange.emit(event);
  }

  handleOnCalculationControlCompFieldRemoved(event) {
    this.onCalculationControlCompFieldRemoved.emit(event);
  }

  handleOnCalculationControlCompFieldAdded(event) {
    this.onCalculationControlCompFieldAdded.emit(event);
  }

  // Rich Text pass through methods
  handleOnRichTextControlContentChange(event) {
    this.onRichTextControlContentChange.emit(event);
  }

  // Chart pass through methods
  handleChartControlSettingsClick() {
    this.onChartControlToggleSettingsPanelClick.emit();
  }

  // Image pass though methods
  handleSaveImage(event) {
    this.onSaveImage.emit(event);
  }

  handleRemoveImage(deleteImageRequest) {
    this.onRemoveImage.emit(deleteImageRequest);
  }
}
