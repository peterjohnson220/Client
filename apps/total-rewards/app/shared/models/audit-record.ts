export interface AuditRecord {
  $type: string;
  EditedById: number;
  EditedBy: string;
  EditedDateTime: Date;
  CreatedById: number;
  CreatedBy: string;
  CreatedDateTime: Date;
  CompanyId: 13;
}

export function generateMockAuditRecord(): AuditRecord {
  return {
    $type: 'Payfactors.Domain.TotalRewards.Audit.EditAuditRecordDto, Payfactors.Domain',
    EditedById: 12689,
    EditedBy: 'edited by',
    EditedDateTime: new Date('2020-04-15T20:06:39.817Z'),
    CreatedById: 12689,
    CreatedBy: 'created by',
    CreatedDateTime: new Date('2020-04-13T17:54:21.42Z'),
    CompanyId: 13,
  };
}
