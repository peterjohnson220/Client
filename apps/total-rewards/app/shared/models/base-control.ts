import { Layout } from './layout';
import { TotalRewardsControlEnum } from './total-rewards-control-enum';
import {LabelWithOverride} from './label-with-override';

export interface BaseControl {
  $type: string; // this is used for serialization purposes: https://www.newtonsoft.com/json/help/html/SerializeTypeNameHandling.htm
  Id: string;
  Title: LabelWithOverride;
  ControlType: TotalRewardsControlEnum;
  Layout: Layout;
}
