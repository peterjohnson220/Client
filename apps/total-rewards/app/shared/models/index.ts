export * from './statement';
export * from './compensation-field';
export * from './statement-list-view-model';
export * from './page';
export * from './total-rewards-control-enum';
export * from './base-control';
export * from './calculation-control';
export * from './chart-control';
export * from './column';
export * from './image-control';
export * from './layout';
export * from './section';
export * from './rich-text-control';
export * from './rich-text-data-field';
export * from './title-control';
export * from './request-models';
export * from './label-with-override';
export * from './employee-rewards-data';
export * from './statement-mode-enum';
export * from './compensation-field';
export * from './audit-record';
export * from './settings';
