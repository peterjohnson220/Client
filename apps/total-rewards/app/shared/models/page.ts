import { Section } from './section';

export interface Page {
  Sections: Section[];
}
