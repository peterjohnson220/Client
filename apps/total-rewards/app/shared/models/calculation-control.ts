import { BaseControl } from './base-control';
import { CompensationField } from './compensation-field';
import { LabelWithOverride } from './label-with-override';
import { TotalRewardsControlEnum } from './total-rewards-control-enum';

export interface CalculationControl extends BaseControl {
  Category: string;
  DataFields: CompensationField[];
  Summary: LabelWithOverride;
}

export function generateMockCalculationControl(): CalculationControl {
  return {
    Id: '105',
    $type: 'TotalRewardsCalculationControlDto',
    Title: {Default: 'Cash Compensation', Override: ''},
    ControlType: TotalRewardsControlEnum.Calculation,
    Layout: { Width: 12 },
    Category: 'Compensation',
    DataFields: [
      { Id: '1', DatabaseField: 'Base', Name: {Default: 'Base Salary', Override: ''}, IsVisible: true},
      { Id: '2', DatabaseField: 'Bonus', Name: {Default: 'Bonus', Override: ''}, IsVisible: true},
      { Id: '3', DatabaseField: 'STI', Name: {Default: 'Short Term Incentive', Override: ''}, IsVisible: true}
    ],
    Summary: {Default: '', Override: ''}
  };
}
