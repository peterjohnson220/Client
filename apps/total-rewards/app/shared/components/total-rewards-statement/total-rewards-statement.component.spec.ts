import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { DragulaModule } from 'ng2-dragula';
import { provideMockStore, MockStore } from '@ngrx/store/testing';

import { generateDefaultAsyncStateObj } from 'libs/models';

import * as statementEditReducer from '../../../_main/statement-edit/reducers/statement-edit.page.reducer';
import * as fromTotalRewardsStatementEditReducer from '../../../_main/statement-edit/reducers';
import {
  Statement,
  generateMockStatement,
  generateMockStatementWithSingleControl,
  generateMockEmployeeRewardsData,
  generateMockStatementWithSingleCalculationControl,
  generateMockStatementWithSingleCalculationControlAndNoVisibleFields
} from '../../models';
import { TotalRewardsControlEnum } from '../../models';
import { TotalRewardsStatementComponent } from './total-rewards-statement.component';

describe('TotalRewardsStatementComponent', () => {
  let component: TotalRewardsStatementComponent;
  let fixture: ComponentFixture<TotalRewardsStatementComponent>;

  const initialState = { totalRewards_statementEdit: { page: statementEditReducer.initialState } };
  initialState.totalRewards_statementEdit.page.statement = generateDefaultAsyncStateObj<Statement>(generateMockStatement());
  let store: MockStore<fromTotalRewardsStatementEditReducer.State>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [DragulaModule.forRoot()],
      providers: [
        provideMockStore({})
      ],
      declarations: [TotalRewardsStatementComponent],
      schemas: [NO_ERRORS_SCHEMA]
    });

    store = TestBed.inject(MockStore);
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TotalRewardsStatementComponent);
    component = fixture.componentInstance;
  });

  it('should create with initial state', () => {
    // arrange
    store.setState({ totalRewards_statementEdit: { page: statementEditReducer.initialState } } as any);

    // act
    fixture.detectChanges();

    // assert
    expect(component).toBeTruthy();
  });

  it('should render a mock statement derived from template A', () => {
    // arrange
    fixture.componentInstance.statement = statementEditReducer.initialState.statement.obj;
    component.employeeRewardsData = generateMockEmployeeRewardsData();

    // act
    fixture.detectChanges();

    // assert
    expect(fixture).toMatchSnapshot();
  });

  it('should render the expected number of pages', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj.Pages = [{} as any, {} as any];
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const pages = fixture.nativeElement.querySelectorAll('.trs-page');
    expect(pages.length).toBe(2);
  });

  it('should render the expected number of sections', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj.Pages = [{ Sections: [{}, {}, {}, {}, {}] }] as any;
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const sections = fixture.nativeElement.querySelectorAll('section.trs-section');
    expect(sections.length).toBe(5);
  });

  it('should render the expected number of columns', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj.Pages = [{ Sections: [{ Columns: [{}] } as any ] }];
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const columns = fixture.nativeElement.querySelectorAll('.trs-column');
    expect(columns.length).toBe(1);
  });

  it('should render a rich text control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleControl(TotalRewardsControlEnum.RichTextEditor);
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const richTextEditors = fixture.nativeElement.querySelectorAll('pf-trs-rich-text-control');
    expect(richTextEditors.length).toBe(1);
  });

  it('should render an image control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleControl(TotalRewardsControlEnum.Image);
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const imageControls = fixture.nativeElement.querySelectorAll('pf-trs-image-control');
    expect(imageControls.length).toBe(1);
  });

  it('should render a calculation control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleCalculationControl();
    fixture.componentInstance.statement = statementEditState.statement.obj;
    component.employeeRewardsData = generateMockEmployeeRewardsData();

    // act
    fixture.detectChanges();

    // assert
    const calculationControls = fixture.nativeElement.querySelectorAll('pf-trs-calculation-control');
    expect(calculationControls.length).toBe(1);
  });

  it('should render a chart control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleControl(TotalRewardsControlEnum.Chart);
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const chartControls = fixture.nativeElement.querySelectorAll('pf-trs-chart-control');
    expect(chartControls.length).toBe(1);
  });

  it('should render a title control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleControl(TotalRewardsControlEnum.Title);
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const titleControls = fixture.nativeElement.querySelectorAll('pf-trs-title-control');
    expect(titleControls.length).toBe(1);
  });

  it('should render a summary control', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleControl(TotalRewardsControlEnum.CalculationSummary);
    fixture.componentInstance.statement = statementEditState.statement.obj;

    // act
    fixture.detectChanges();

    // assert
    const summaryControls = fixture.nativeElement.querySelectorAll('pf-trs-summary-control');
    expect(summaryControls.length).toBe(1);
  });

  it('should not include calc control if no rewards data', () => {
    // arrange
    component.statement = generateMockStatement();
    component.employeeRewardsData = generateMockEmployeeRewardsData();
    component.employeeRewardsData.EmployeeBase = null;
    component.employeeRewardsData.EmployeeBonus = null;
    component.employeeRewardsData.EmployeeSTI = null;
    component.employeeRewardsData.EmployeeLTI = null;

    // act
    fixture.detectChanges();

    // assert
    expect(component.visibleCalculationControls.length).toBe(3);
  });

  it('should not include calc control if rewards data is present, but no fields are visible', () => {
    // arrange
    const statementEditState = statementEditReducer.initialState;
    statementEditState.statement.obj = generateMockStatementWithSingleCalculationControlAndNoVisibleFields();
    component.statement = statementEditState.statement.obj;

    component.employeeRewardsData = generateMockEmployeeRewardsData();

    // act
    fixture.detectChanges();

    // assert
    const calculationControls = fixture.nativeElement.querySelectorAll('pf-trs-calculation-control');
    expect(calculationControls.length).toBe(0);
  });
});
