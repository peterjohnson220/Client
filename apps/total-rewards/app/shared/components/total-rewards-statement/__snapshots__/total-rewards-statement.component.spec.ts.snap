// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalRewardsStatementComponent should render a mock statement derived from template A 1`] = `
<pf-total-rewards-statement
  controlType={[Function Object]}
  employeeArray={[Function Array]}
  employeeData={[Function Array]}
  onCalculationControlCompFieldAdded={[Function EventEmitter]}
  onCalculationControlCompFieldRemoved={[Function EventEmitter]}
  onCalculationControlCompFieldTitleChange={[Function EventEmitter]}
  onCalculationControlSummaryTitleChange={[Function EventEmitter]}
  onChartControlToggleSettingsPanelClick={[Function EventEmitter]}
  onControlTitleChange={[Function EventEmitter]}
  onRemoveImage={[Function EventEmitter]}
  onRichTextControlContentChange={[Function EventEmitter]}
  onSaveImage={[Function EventEmitter]}
  pageCount={[Function Number]}
  pageCountSubscription={[Function Subscriber]}
  route={[Function Object]}
  statement={[Function Object]}
  statementModeEnum={[Function Object]}
>
  <div
    class="container statement mb-4 rounded medium-font-size default-font-family"
    ng-reflect-klass="container statement mb-4 round"
    ng-reflect-ng-class="medium-font-size default-font-"
  >
    
    <div>
      
      <div
        class="pb-4 trs-page"
      >
        
        <section
          class="row vertical-divider trs-section"
        >
          
          <div
            class="trs-column col-6"
            ng-reflect-klass="trs-column"
            ng-reflect-ng-class="col-6"
          >
            <div
              class="row columnRow"
            >
              
              <div
                class="col-12 control-type-2"
                ng-reflect-ng-class="col-12 control-type-2"
              >
                
                
                <div>
                  <pf-trs-image-control
                    class="p-1"
                  />
                </div>
                
                
                
                
              </div>
            </div>
          </div>
          <div
            class="trs-column col-6"
            ng-reflect-klass="trs-column"
            ng-reflect-ng-class="col-6"
          >
            <div
              class="row columnRow"
            >
              
              <div
                class="col-12 control-type-5"
                ng-reflect-ng-class="col-12 control-type-5"
              >
                
                
                
                
                
                <div>
                  <pf-trs-title-control
                    class="p-1"
                  />
                </div>
                
              </div>
            </div>
          </div>
        </section>
        <section
          class="row vertical-divider trs-section"
        >
          
          <div
            class="trs-column col-6"
            ng-reflect-klass="trs-column"
            ng-reflect-ng-class="col-6"
          >
            <div
              class="row columnRow"
            >
              
              <div
                class="col-12 control-type-1"
                ng-reflect-ng-class="col-12 control-type-1"
              >
                
                <div>
                  <pf-trs-rich-text-control
                    class="p-1"
                  />
                </div>
                
                
                
                
                
              </div>
              <div
                class="col-12 control-type-4"
                ng-reflect-ng-class="col-12 control-type-4"
              >
                
                
                
                
                <div>
                  <pf-trs-chart-control
                    class="p-1"
                  />
                </div>
                
                
              </div>
              <div
                class="col-12 control-type-1"
                ng-reflect-ng-class="col-12 control-type-1"
              >
                
                <div>
                  <pf-trs-rich-text-control
                    class="p-1"
                  />
                </div>
                
                
                
                
                
              </div>
            </div>
          </div>
          <div
            class="trs-column col-6"
            ng-reflect-klass="trs-column"
            ng-reflect-ng-class="col-6"
          >
            <div
              class="row columnRow"
            >
              
              <div
                class="col-12 control-type-3"
                ng-reflect-ng-class="col-12 control-type-3"
              >
                
                
                
                <div>
                  <pf-trs-calculation-control />
                </div>
                
                
                
              </div>
              <div
                class="col-12 control-type-3"
                ng-reflect-ng-class="col-12 control-type-3"
              >
                
                
                
                <div>
                  <pf-trs-calculation-control />
                </div>
                
                
                
              </div>
              <div
                class="col-12 control-type-3"
                ng-reflect-ng-class="col-12 control-type-3"
              >
                
                
                
                <div>
                  <pf-trs-calculation-control />
                </div>
                
                
                
              </div>
              <div
                class="col-12 control-type-3"
                ng-reflect-ng-class="col-12 control-type-3"
              >
                
                
                
                <div>
                  <pf-trs-calculation-control />
                </div>
                
                
                
              </div>
              <div
                class="col-12 control-type-6"
                ng-reflect-ng-class="col-12 control-type-6"
              >
                
                
                
                
                
                
                <div>
                  <pf-trs-summary-control
                    class="p-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
      <div
        class="page-break"
      />
    </div>
  </div>
</pf-total-rewards-statement>
`;
